src/common/color-element.js

```diff
@@ -137,6 +140,42 @@ const Self = class ColorElement extends NudeElement {
 
 		customElements.define(this.tagName, this);
 	}
+
+	/**
+	 * Resolve a color value and cache it.
+	 * @param {string} value Color value to resolve.
+	 * @param {Element} element Element to get computed style from to resolve the color value.
+	 * @returns {Color | null} Resolved color value or null if the value cannot be resolved.
+	 */
+	static resolveColor (value, element) {
+		try {
+			return Self.Color.get(value);
+		}
+		catch {}
+
+		// Color.js can't parse the color value; possibly one of the values we can handle gracefully
+		if (Self.#resolvedColors.has(value)) {
+			let color = Self.#resolvedColors.get(value);
+			return Self.Color.get(color);
+		}
+
+		if (!globalThis.CSS?.supports("color", value) || !element) {
+			// Not supported/invalid value, or no element to resolve the color value from
+			return null;
+		}
+
+		// One of the supported color values; resolve and cache it
+		element.style.backgroundColor = value;
```

Comment:
We should store the element's previous `backgroundColor` so we can restore it, rather than overwriting.
Also, we should avoid the DOM I/O if the element *already* has that background (since any DOM operation is slow), so perhaps we can pass in an option for that.

Reply:
> We should store the element's previous `backgroundColor` so we can restore it, rather than overwriting.

I fixed it. Thanks!

> Also, we should avoid the DOM I/O if the element _already_ has that background (since any DOM operation is slow), so perhaps we can pass in an option for that.

Could you elaborate, please? How do you see it?

Reply:
> Could you elaborate, please? How do you see it?

Like, a setting (name TBD) that will tell the function "the element I'm giving you already has the right background, don't bother setting it"